src/github/index.js

```diff
@@ -6,273 +6,243 @@ import logger from '../logger/index.js';
-    // but it goes with a 404 using octokit
-    // baseUrl: `https://api.github.com/${GITHUB_OTA_OWNER}/${GITHUB_OTA_REPO}`,
-    return items.filter(item => item.repository_url.endsWith(`${params.owner}/${params.repo}`) && item.title === title);
-  } catch (e) {
-    logger.error('Could not search issue');
-    logger.error(e.toString());
-    throw e;
   }
-};
 
-export const addCommentToIssue = async params => {
-  if (!isEnabled) {
-    return;
-  }
-  try {
-    const { data } = await octokit.rest.issues.createComment(params);
+  async addCommentToIssue(params) {
+    try {
+      const { data } = await this.octokit.rest.issues.createComment(params);
 
-    return data;
-  } catch (e) {
-    logger.error('Could not add comment to issue');
-    logger.error(e.toString());
+      return data;
+    } catch (e) {
+      logger.error('Could not add comment to issue:', e.toString());
 
-    return null;
+      return null;
+    }
   }
-};
 
-export const createIssueIfNotExist = async ({ title, body, labels, comment }) => {
-  if (!isEnabled) {
-    return;
-  }
+  async createIssueIfNotExist({ title, body, labels, comment }) {
+    try {
+      const existingIssues = await this.searchIssues({ ...this.commonParams, title, q: `is:issue label:${labels.join(',')}` });
 
-  try {
-    const existingIssues = await searchIssues({
-      ...commonParams,
-      title,
-      q: `is:issue label:${labels.join(',')}`,
-    });
+      if (!existingIssues.length) {
+        const existingIssue = await this.createIssue({ ...this.commonParams, title, body, labels });
+
+        if (existingIssue) {
+          logger.info(`🤖 Creating Github issue for ${title}: ${existingIssue.html_url}`);
+        } else {
+          logger.error(`🤖 Could not create Github issue for ${title}`);
+        }
 
-    if (!existingIssues[0]) {
-      const existingIssue = await createIssue({
-        ...commonParams,
-        title,
-        body,
-        labels,
-      });
-
-      if (existingIssue) {
-        logger.info(`🤖 Creating Github issue for ${title}: ${existingIssue.html_url}`);
-      } else {
-        logger.error(`🤖 Could not create Github issue for ${title}`);
+        return existingIssue;
       }
 
-      return existingIssue;
+      const openedIssues = existingIssues.filter(existingIssue => existingIssue.state === ISSUE_STATE_OPEN);
+      const hasNoneOpened = openedIssues.length === 0;
+
+      for (const existingIssue of existingIssues) {
+        if (hasNoneOpened) {
+          await this.octokit.rest.issues.update({ ...this.commonParams, issue_number: existingIssue.number, state: ISSUE_STATE_OPEN }); // eslint-disable-line no-await-in-loop
+          await this.addCommentToIssue({ ...this.commonParams, issue_number: existingIssue.number, body: comment }); // eslint-disable-line no-await-in-loop
+          logger.info(`🤖 Reopened automatically as an error occured for ${title}: ${existingIssue.html_url}`);
+          break;
```

Comment:
If we `break` on the first iteration, why exactly are we in a `for` loop? 🤔

This whole `filter` / `for` / `break` seems functionally equivalent to `existingIssue = existingIssues.find(existingIssue => existingIssue.state === ISSUE_STATE_OPEN)`.